pkg/devicetwin/devicetwin_test.go

```diff
@@ -0,0 +1,193 @@
+	core.Register(&DeviceTwin{})
+	dt := DeviceTwin{}
+	mainContext.AddModule(dt.Name())
+	mainContext.AddModuleGroup(dt.Name(), dt.Group())
+	dt.context = mainContext
+	controller, err := InitDTController(mainContext)
+	if err != nil {
+		t.Errorf("Start device twin failed, due to %v", err)
+	}
+	dt.dtcontroller = controller
+	ormerMock.EXPECT().QueryTable(gomock.Any()).Return(querySeterMock).Times(200)
+	querySeterMock.EXPECT().All(gomock.Any()).Return(int64(1), nil).Times(200)
+	go dt.Start(mainContext)
+	// Sending a message from devicetwin module to the created fake module(TestModule).
+	msg := model.NewMessage("").BuildRouter(TestModule, GroupResource, model.ResourceTypePodStatus, model.InsertOperation)
+	dt.context.Send(TestModule, *msg)
+	_, err = mainContext.Receive(TestModule)
+	t.Run("MessagePingTest", func(t *testing.T) {
+		if err != nil {
+			t.Errorf("Error while receiving message: %v", err)
+			return
+		}
+	})
+	//Checking health of MemModule
+	dtmName := dtcommon.MemModule
+	t.Run("MemModuleHealthCheck", func(t *testing.T) {
+		health, ok := controller.DTContexts.ModulesHealth.Load(dtmName)
+		if ok {
+			now := time.Now().Unix()
+			if now-health.(int64) > 60*2 {
+				t.Errorf("%s health %v is old", dtmName, health)
+			}
+		}
+	})
+	//Checking health of TwinModule
+	dtmName = dtcommon.TwinModule
+	t.Run("TwinModuleHealthCheck", func(t *testing.T) {
+		health, ok := controller.DTContexts.ModulesHealth.Load(dtmName)
+		if ok {
+			now := time.Now().Unix()
+			if now-health.(int64) > 60*2 {
+				t.Errorf("%s health %v is old", dtmName, health)
+			}
+		}
+	})
+	//Checking health of DeviceModule
+	dtmName = dtcommon.DeviceModule
+	t.Run("DeviceModuleHealthCheck", func(t *testing.T) {
+		health, ok := controller.DTContexts.ModulesHealth.Load(dtmName)
+		if ok {
+			now := time.Now().Unix()
+			if now-health.(int64) > 60*2 {
+				t.Errorf("%s health %v is old", dtmName, health)
+			}
+		}
+	})
+	//Checking health of CommModule
+	dtmName = dtcommon.CommModule
+	t.Run("CommModuleHealthCheck", func(t *testing.T) {
+		health, ok := controller.DTContexts.ModulesHealth.Load(dtmName)
+		if ok {
+			now := time.Now().Unix()
+			if now-health.(int64) > 60*2 {
+				t.Errorf("%s health %v is old", dtmName, health)
+			}
+		}
+	})
+}
+
+// TestCleanup is function to test Cleanup().
+func TestCleanup(t *testing.T) {
+	var deviceTwin DeviceTwin
+	deviceTwin.context = context.GetContext(context.MsgCtxTypeChannel)
+	nodeid, _ := config.CONFIG.GetValue("edgehub.controller.node-id").ToString()
+	deviceTwin.dtcontroller = &DTController{
+		HeartBeatToModule: make(map[string]chan interface{}),
+		DTContexts: &dtcontext.DTContext{
+			GroupID:     "",
+			NodeID:      nodeid,
+			CommChan:    make(map[string]chan interface{}),
+			ConfirmChan: make(chan interface{}, 1000),
+		},
+		DTModules: make(map[string]dtmodule.DTModule),
+		Stop:      make(chan bool, 1),
+	}
+	t.Run("cleanuptest", func(t *testing.T) {
+		dt := &deviceTwin
+		dt.Cleanup()
```

Comment:
I don't see any checks in this function. Please check https://github.com/kubeedge/kubeedge/blob/master/pkg/metamanager/module_test.go to see how cleanup is tested.